cms/djangoapps/contentstore/tasks.py

```diff
@@ -125,3 +152,222 @@ def push_course_update_task(course_key_string, course_subscription_id, course_di
+
+        def get_dir_for_filename(directory, filename):
+            """
+            Returns the directory path for the first file found in the directory
+            with the given name.  If there is no file in the directory with
+            the specified name, return None.
+            """
+            for name, directory_path in get_all_files(directory):
+                if name == filename:
+                    return directory_path
+            return None
+
+        dirpath = get_dir_for_filename(course_dir, root_name)
+        if not dirpath:
+            with respect_language(language):
+                self.status.fail(_(u'Could not find the {0} file in the package.').format(root_name))
+                return
+
+        dirpath = os.path.relpath(dirpath, data_root)
+        LOGGER.debug(u'found %s at %s', root_name, dirpath)
+
+        LOGGER.info(u'Course import %s: Extracted file verified', courselike_key)
+        self.status.set_state(u'Updating')
+        self.status.increment_completed_steps()
+
+        with dog_stats_api.timer(
+            u'courselike_import.time',
+            tags=[u"courselike:{}".format(courselike_key)]
+        ):
+            courselike_items = import_func(
+                modulestore(), user.id,
+                settings.GITHUB_REPO_ROOT, [dirpath],
+                load_error_modules=False,
+                static_content_store=contentstore(),
+                target_id=courselike_key
+            )
+
+        new_location = courselike_items[0].location
+        LOGGER.debug(u'new course at %s', new_location)
+
+        LOGGER.info(u'Course import %s: Course import successful', courselike_key)
+        self.status.increment_completed_steps()
+    except Exception as exception:   # pylint: disable=broad-except
+        LOGGER.exception(u'error importing course')
+        self.status.fail(text_type(exception))
+    finally:
+        if course_dir.isdir():  # pylint: disable=no-value-for-parameter
+            shutil.rmtree(course_dir)
+            LOGGER.info(u'Course import %s: Temp data cleared', courselike_key)
+
+        if self.status.completed_steps == 3 and root_name == COURSE_ROOT:
+            # Reload the course so we have the latest state
+            course = modulestore().get_course(courselike_key)
+            if course.entrance_exam_enabled:
+                entrance_exam_chapter = modulestore().get_items(
+                    course.id,
+                    qualifiers={u'category': u'chapter'},
+                    settings={u'is_entrance_exam': True}
+                )[0]
+
+                metadata = {u'entrance_exam_id': text_type(entrance_exam_chapter.location)}
+                CourseMetadata.update_from_dict(metadata, course, user)
+                from contentstore.views.entrance_exam import add_entrance_exam_milestone
+                add_entrance_exam_milestone(course.id, entrance_exam_chapter)
+                LOGGER.info(u'Course %s Entrance exam imported', course.id)
```

Comment:
Sure, I'll write a ticket to come back and clean that up (maybe as part of the import refactoring to dump XML modulestore).